### PR TITLE
test(billing): verify plan limit parity between SaaS and self-hosted

### LINKED ISSUE #3353: audit(billing): verify plan limit parity between SaaS subscription and self-hosted license
## Context

LangWatch runs as both SaaS (Stripe subscription) and self-hosted (license key). The `CompositePlanProvider` resolves the active plan: license takes precedence over subscription, falls back to FREE.

We need to verify that **the same enforcement logic applies regardless of plan source** — a self-hosted instance with a license should hit the same guards as a SaaS subscriber.

## Problem

The plan resolution pipeline is:
1. `LicensePlanProvider` — reads license key → returns `PlanInfo` with `planSource: "license"`
2. `SaasSubscriptionPlanProvider` — reads Stripe subscription → returns `PlanInfo` with `planSource: "subscription"`
3. `CompositePlanProvider` — picks license if non-free, else subscription, else FREE

### Areas to investigate

- **License plan shape**: Does a license-sourced `PlanInfo` populate ALL limit fields (`maxWorkflows`, `maxPrompts`, etc.) or only a subset? If fields are missing/undefined, enforcement may silently pass.
- **`overrideAddingLimitations`**: This flag lets admins/impersonators bypass limits. Verify it's correctly computed for both license and subscription flows.
- **License expiry**: When a license expires, does the system fall back to FREE or to the SaaS subscription? What happens to resources created under the higher limits?
- **Edge cases**: Organization with BOTH a valid license AND an active subscription — the composite provider picks the license. Is there a scenario where subscription limits are more generous and the user gets downgraded unexpectedly?
- **Self-hosted without license**: Default behavior should be FREE plan with full enforcement. Verify no code path assumes self-hosted = unlimited.

## Acceptance Criteria

- [ ] Document the current behavior of `CompositePlanProvider` for all combinations (license only, subscription only, both, neither, expired license)
- [ ] Verify all 15 limit types are populated in both license-sourced and subscription-sourced `PlanInfo`
- [ ] Add integration tests for license + subscription interaction edge cases
- [ ] Fix any gaps where license-sourced plans skip enforcement that subscription plans enforce
- [ ] Verify `overrideAddingLimitations` behavior is consistent across both sources

## References

- Composite provider: `server/app-layer/subscription/composite-plan-provider.ts`
- License provider: `ee/licensing/`
- Plan limits: `ee/billing/planLimits.ts`
- Audit doc: `EPIC/Billing/limits.md` (Obsidian)

### PR DESCRIPTION
**Pull request by sergioestebance**
## Summary
- Add BDD feature spec (`specs/licensing/plan-limit-parity.feature`) documenting 13 scenarios for composite plan provider behavior
- Add unit tests verifying all 16 numeric limit fields are populated for both SaaS (PLAN_LIMITS) and license-sourced plans (mapToPlanInfo + resolvePlanDefaults)
- Add unit tests for composite provider edge cases: license-first precedence when subscription is more generous, both-free fallback, field completeness
- Add `@scenario` annotations binding all 13 feature scenarios to their test implementations
- Verify `overrideAddingLimitations` is consistently recomputed from user context regardless of plan source

## Test plan
- [x] `planLimits.unit.test.ts` — 16 numeric field completeness for every SaaS plan type + license defaults
- [x] `composite-plan-provider.unit.test.ts` — selection logic, override consistency, edge cases
- [x] Feature-parity check: 13/13 scenarios bound
- [x] All unit and integration tests pass in CI
- [x] Typecheck passes (no errors in changed files)

Closes #3353

### CODE REVIEW

File: langwatch/ee/billing/__tests__/planLimits.unit.test.ts [hunk 1]
```diff
@@ -16,4 +43,79 @@ describe("PLAN_LIMITS", () => {
...
+      /** @scenario SaaS-sourced plan populates all limit fields */
+      it.each(planEntries)(
+        "populates all 16 numeric limit fields for %s",
+        (_planType, plan) => {
+          for (const field of NUMERIC_LIMIT_FIELDS) {
+            const value = plan[field];
+            expect(value, `${_planType}.${field} is undefined`).toBeDefined();
+            expect(typeof value, `${_planType}.${field} is not a number`).toBe(
+              "number",
+            );
```
[the remark this answers is not in the document]
Comment (sergioestebance) [reply langwatch/ee/billing/__tests__/planLimits.unit.test.ts]: Good catch — applied `Number.isFinite()` to both loops. `DEFAULT_LIMIT` (`Number.MAX_SAFE_INTEGER`) is finite so the license defaults test still passes.
